code/go/0chain.net/smartcontract/storagesc/blobber.go

```diff
@@ -286,72 +260,77 @@ func (sc *StorageSmartContract) updateBlobberSettings(t *transaction.Transaction
 			"access denied, allowed for delegate_wallet owner only")
 	}
 
-	// stake pool settings
+	// upate some extra props
+	blobber.Terms = updatedBlobber.Terms
+	blobber.Capacity = updatedBlobber.Capacity
 
-	if err = update.StakePoolSettings.validate(conf); err != nil {
-		return "", common.NewError("update_blobber_settings_failed",
-			"validating new stake pool settings: "+err.Error())
-	}
-
-	sp.Settings.MinStake = update.StakePoolSettings.MinStake
-	sp.Settings.MaxStake = update.StakePoolSettings.MaxStake
-	sp.Settings.ServiceCharge = update.StakePoolSettings.ServiceCharge
-	sp.Settings.NumDelegates = update.StakePoolSettings.NumDelegates
-
-	// blobber settings
-	blob.Terms = update.Terms
-	blob.Capacity = update.Capacity
-
-	if blob.Capacity == 0 {
-		// if already removed
-		if update.Capacity == 0 {
-			// keep removed
-		} else {
-			// reborn
-			sc.statIncr(statNumberOfBlobbers)
-			all.Nodes.add(blob)
-		}
-		sc.statIncr(statUpdateBlobber)
-	} else {
-		// alive blobber
-		if update.Capacity == 0 {
-			// remove
-			all.Nodes.remove(blob.ID)
-			sc.statIncr(statRemoveBlobber)
-			sc.statDecr(statNumberOfBlobbers)
-		} else {
-			// keep alive
-			all.Nodes.add(blob)
-			sc.statIncr(statUpdateBlobber)
-		}
+	// update blobber
+	if err = sc.updateBlobber(t, conf, blobber, blobbers, balances); err != nil {
+		return "", common.NewError("update_blobber_settings_failed", err.Error())
 	}
```

Comment:
This is an example of bad comment i am talking about, the other comments similar but its just not this bad.

Reply:
dropped it.